pkg/collector/adapters/config_to_ports_test.go

```diff
@@ -144,6 +144,39 @@ service:
 	}
 }
 
+func TestReceiverPortsSortedCorrectly(t *testing.T) {
+	configurationString := `receivers:
+  otlp:
+    protocols:
+      grpc:
+      http:  
+  jaeger:
+    protocols:
+      grpc:
+service:
+  pipelines:
+    traces:
+      receivers: [otlp, jaeger]
+      exporters: [logging]
+`
+	configuration, err := adapters.ConfigFromString(configurationString)
+	require.NoError(t, err)
+	require.NotEmpty(t, configuration)
+
+	ports, err := adapters.ConfigToReceiverPorts(logger, configuration)
+	assert.NoError(t, err)
+	assert.Len(t, ports, 4)
```

Comment:
This PR changes the semantics of `ConfigToReceiverPorts` - the expected result is strictly ordered, hence we can write a nice assert and compare slices directly:

```golang
assert.Equal(error, []v1.port{{Name: "foo", Port: 1778944}{...}}, got)
```